app/controllers/home_controller.rb

```diff
@@ -20,13 +21,15 @@ def create
   def show
     @link = Link.find(params[:id])
     @comment = Comment.new
-    @comments = Comment.all
+    @comments = @link.comments.all
     flash[:back] = home_path(params[:id])
   end
 
   def comment_create
+    @link = Link.find(params[:comment][:commentable])
     @comment = Comment.new(params_comment)
```

Comment:
can we use Link.find_by(id: ) so that it will not give exception when item is not present!

Reply:
sure, the params is comming from a hidden field from show params, so it shouldn't be possible

Reply:
sure, the `params[:comment][:commentable] ` is from a hidden_field so it should always be present but i will replace it wth find_by just in case